coreos-installer

```diff
@@ -378,6 +378,42 @@ write_networking_opts() {
     echo "set ignition_network_kcmdline=\"$(cat /tmp/networking_opts)\"" >> /mnt/boot_partition/ignition.firstboot
 }
 
+#########################################################
+# Helper to persist ignition platform id to initramfs first boot
+#########################################################
+write_platform_id() {
+    local tmp_platform_id="/tmp/platform_id"
+
+    if [ ! -f ${tmp_platform_id} ]; then
+        log "Not overwriting ignition platform id, no platform id provided"
+        return
+    else
+        local platform_id="$(cat ${tmp_platform_id})"
+    fi
+
+
+    log "Overwriting ignition platform id"
+    # check for the boot partition
+    mkdir -p /mnt/boot_partition
+    mount_boot_partition /mnt/boot_partition
+    trap 'umount /mnt/boot_partition; trap - RETURN' RETURN
+    local conf_files=$(ls /mnt/boot_partition/loader/entries/*conf)
+
+    for f in $conf_files
+    do
+        if grep -q -E "ignition.platform.id=[^$ ]*" $f; then
+            # ignition.platform.id is followed by whitespace or EOF
+            sed -i "s/ignition.platform.id=[^$ ]*/ignition.platform.id=${platform_id}/" $f
+        else
+            # ignition.platform.id is not presented in BLS
+            sed -i "/^options/ s/$/ ignition.platform.id=${platform_id}" $f
```

Comment:
does this work? I get an error:

```
[dustymabe@media tmp]$ cat new.conf 
options root=/dev/disk/by-label/root rootflags=defaults,prjquota rw $ignition_firstboot mitigations=auto,nosmt console=tty0 console=ttyS0,115200n8 ostree=/ostree/boot.1/fedora-coreos/030433db74a4e975020ed5c806c9e4e066665480e05f43463024bfde3b90e02e/0
[dustymabe@media tmp]$ sed "/^options/ s/$/ ignition.platform.id=${platform_id}" new.conf 
sed: -e expression #1, char 37: unterminated `s' command
```

Reply:
Looks like it's missing a trailing slash.

Reply:
It should be `sed -i "/^options/ s/$/ ignition.platform.id=${platform_id}/" $f`, I'm missing a `/` after `${platform_id}`